### ISSUE TITLE
how to use Indicators with custom_stoploss

### ISSUE THREAD

**Issue by hippocritical** [freqtrade/freqtrade#4793]
## Describe your environment

  * Operating system: windows
  * Python Version: docker-container
  * CCXT version: 1.48.22
  * Freqtrade Version: latest x64 develop_plot version (24.04.2021)
  
## Your question

Hi, I am currently trying to do a custom stoploss depending on the ATR indicator.
Simply said, I just want to calculate the percentage from the current price to the ATR in percent and use that as my starting point.

when the backtesting started this error appears. It disappears once a static value is returned (logically).

**2021-04-24 12:04:36,331 - freqtrade.strategy.strategy_wrapper - ERROR - Unexpected error name 'dataframe' is not defined calling <bound method My_Ema_bullish_confirmation.custom_stoploss of <My_Ema_bullish_confirmation.My_Ema_bullish_confirmation object at 0x7fb7b4d55ac0>>
Traceback (most recent call last):
  File "/freqtrade/freqtrade/strategy/strategy_wrapper.py", line 17, in wrapper
    return f(*args, **kwargs)
  File "/freqtrade/user_data/strategies/My_Ema_bullish_confirmation.py", line 32, in custom_stoploss
    return -1 * abs(dataframe['atr'] / dataframe['open'])
NameError: name 'dataframe' is not defined
2021-04-24 12:04:36,334 - freqtrade - ERROR - name 'dataframe' is not defined**

This is the code snippet:

`class My_Ema_bullish_confirmation(IStrategy):
    # Minimal ROI designed for the strategy.
    # This attribute will be overridden if the config file contains "minimal_roi"
    custom_info = {}
    
    minimal_roi = {"0": 0.9}
    stoploss = -0.04
    # Trailing stop:
    #trailing_stop = True

    # Optimal timeframe for the strategy
    timeframe = '5m'

    use_custom_stoploss = True
        
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        return -1 * abs(dataframe['atr'] / dataframe['open'])

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe['EMAShort'] = ta.EMA(dataframe, timeperiod=10, price='close')
        dataframe['EMALong'] = ta.EMA(dataframe, timeperiod=20, price='close')
        
        dataframe['EMAShort_3Ago'] = ta.EMA(dataframe, timeperiod=10, price='close').shift(3)
        dataframe['EMAShort_2Ago'] = ta.EMA(dataframe, timeperiod=10, price='close').shift(2)
        dataframe['EMAShort_1Ago'] = ta.EMA(dataframe, timeperiod=10, price='close').shift(1)
        
        dataframe['EMALong_3Ago'] = ta.EMA(dataframe, timeperiod=20, price='close').shift(3)
        dataframe['EMALong_2Ago'] = ta.EMA(dataframe, timeperiod=20, price='close').shift(2)
        dataframe['EMALong_1Ago'] = ta.EMA(dataframe, timeperiod=20, price='close').shift(1)
        
        dataframe['close_3Ago'] = dataframe['close'].shift(3)
        dataframe['close_2Ago'] = dataframe['close'].shift(2)
        dataframe['close_1Ago'] = dataframe['close'].shift(1)
        
        # using "ATR" here as example
        dataframe['atr'] = ta.ATR(dataframe)
        if self.dp.runmode.value in ('backtest', 'hyperopt'):
          # add indicator mapped to correct DatetimeIndex to custom_info
          self.custom_info[metadata['pair']] = dataframe[['date', 'atr']].copy().set_index('date')
        return dataframe`

**Comment by xmatthias** [comment]
Please read the [documentation](https://www.freqtrade.io/en/stable/strategy-advanced/#custom-stoploss-using-an-indicator-from-dataframe-example) which explains how to do this in detail.
Apparently you read some of it (adding it to custom_info) - but not the rest (how to use it in the stoploss).

**Comment by shakeel-appyhigh** [comment]
Hi, has the custom_stoploss function been removed in the current release? I can find the documentation in an earlier release [documentation.](https://www.freqtrade.io/en/2021.2/strategy-advanced/#custom-stoploss-examples) but not in the current release [documentation](https://www.freqtrade.io/en/stable/strategy-advanced)

I have used the following code:
```
    use_custom_stoploss = True
    custom_info = {}
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        
        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        return dataframe['atr']
```
But I get the error `AttributeError: 'RSICrossOverEmaSma' object has no attribute 'stoploss'`

I also tried to store it in custom_info dictionary both as `self.custom_info["stoploss"]` and  `self.custom_info["cust_stoploss"]` but got the same error.

Is the way different in the current release? It is not clear from the documentation how to implement it.
